client/src/App.tsx

```diff
@@ -40,38 +40,54 @@ const App: React.FC = () => {
 	}, []);
 
 	useEffect(() => {
-		fetch("/api/v1/geo/geo-aggregate-data")
-			.then(async (response) => {
-				if (response.ok) {
-					return response.json();
-				} else if (response.status === 404) {
-					const defaultResponse = await fetch(
-						"/api/v1/data-folder/default-simplified.geojson"
-					);
-					const defaultData = await defaultResponse.json();
-					setMapData(defaultData);
-					localStorage.setItem("hasAggregatedData", "false");
-					return null;
-				} else {
-					throw new Error(`Server error: ${response.status}`);
-				}
-			})
-			.then((data) => {
-				if (data) {
+		if (uploadedFiles.length > 0) {
+			fetch("/api/v1/geo/geo-aggregate-data")
+				.then(async (response) => {
+					if (response.ok) {
+						return response.json();
+					} else if (response.status === 404) {
+						throw new Error("No aggregated data found");
+					} else {
+						throw new Error(`Server error: ${response.status}`);
+					}
+				})
+				.then((data) => {
 					setMapData(data);
 					localStorage.setItem("hasAggregatedData", "true");
-					const newIndex = uploadedFiles.length - 1;
-					setCurrentFileIndex(newIndex);
-				}
-			})
-			.catch((error) => {
-				console.error("Failed to load data:", error);
-			});
-	}, [setCurrentFileIndex, uploadCount, uploadedFiles.length]);
+					setCurrentFileIndex(uploadedFiles.length - 1);
+				})
+				.catch((error) => {
+					console.error("Failed to load aggregated data:", error);
+					// If no aggregated data found or error occurs, fetch default data
+					fetch("/api/v1/data-folder/default-simplified.geojson")
+						.then((response) => response.json())
+						.then((defaultData) => {
+							setMapData(defaultData);
+							localStorage.setItem("hasAggregatedData", "false");
+							setCurrentFileIndex(-1); // Reset current file index as no file is selected
+						})
+						.catch((defaultError) => {
+							console.error("Failed to load default GeoJSON data:", defaultError);
+						});
+				});
+		} else {
+			// If no files are uploaded, fetch default data
+			fetch("/api/v1/data-folder/default-simplified.geojson")
```

Comment:
consider making this a const fn within the useEffect that is called on the else block of uploadedFiles.length > 0.